src/plugins/attachments.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2020-present, Elastic NV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { Attachment, Driver, RunOptions, Screenshot } from '../common_types';
+import { log } from '../core/logger';
+import { Step } from '../dsl';
+import { CACHE_PATH, generateUniqueId, getTimestamp } from '../helpers';
+import { mkdir, rm, writeFile } from 'fs/promises';
+import { join } from 'path';
+
+export type AttachmentOptions = {
+  outputDir?: string;
+  screenshots?: RunOptions['screenshots'];
+};
+
+export class AttachmentsManager {
+  private attachments: Attachment[] = [];
+  #screenshotPath = join(CACHE_PATH, 'screenshots');
+
+  constructor(private driver: Driver, private options: AttachmentOptions) {
+    this.#screenshotPath = options.outputDir
```

Comment:
We shouldn't let the user config touch our screenshotpath, otherwise it would break our blocks gen/reconstruction logic. 

The output dir is only taken in to account on the reporter side, where we extract and reconstruct the file paths based on the given value. Thoughts?